components/camel-splunk-hec/src/main/java/org/apache/camel/component/splunkhec/SplunkHECProducer.java

```diff
@@ -0,0 +1,122 @@
+import org.apache.camel.Message;
+import org.apache.camel.support.DefaultProducer;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.EntityTemplate;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.ssl.SSLContextBuilder;
+
+/**
+ * The Splunk HEC producer.
+ */
+public class SplunkHECProducer extends DefaultProducer {
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+    private SplunkHECEndpoint endpoint;
+    private CloseableHttpClient httpClient;
+
+
+    public SplunkHECProducer(SplunkHECEndpoint endpoint) {
+        super(endpoint);
+        this.endpoint = endpoint;
+    }
+
+    @Override
+    protected void doStart() throws Exception {
+        super.doStart();
+        HttpClientBuilder builder = HttpClients.custom().
+                setUserAgent("Camel Splunk HEC/" + getEndpoint().getCamelContext().getVersion()).
+                setMaxConnTotal(10);
+        if (endpoint.getConfiguration().isSkiptlsverify()) {
+            SSLContextBuilder sslbuilder = new SSLContextBuilder();
+            sslbuilder.loadTrustMaterial(null, (chain, authType) -> true);
+            SSLConnectionSocketFactory sslsf = new
+                    SSLConnectionSocketFactory(sslbuilder.build(), NoopHostnameVerifier.INSTANCE);
+            builder.setSSLSocketFactory(sslsf);
+        }
+        httpClient = builder.build();
+    }
+
+    @Override
+    public void process(Exchange exchange) throws Exception {
+        Map<String, Object> payload = createPayload(exchange.getIn());
+
+        HttpPost httppost = new HttpPost((endpoint.getConfiguration().isHttps() ? "https" : "http") + "://" + endpoint.getSplunkURL() + "/services/collector/event");
+        httppost.addHeader("Authorization", " Splunk " + endpoint.getToken());
+
+        EntityTemplate entityTemplate = new EntityTemplate(outputStream -> MAPPER.writer().writeValue(outputStream, payload));
+        entityTemplate.setContentType(ContentType.APPLICATION_JSON.getMimeType());
+
+        httppost.setEntity(entityTemplate);
+        try (CloseableHttpResponse response = httpClient.execute(httppost)) {
+            if (response.getStatusLine().getStatusCode() != 200) {
+                ByteArrayOutputStream output = new ByteArrayOutputStream();
+                response.getEntity().writeTo(output);
+
+                throw new RuntimeException(response.getStatusLine().toString() + "\n" + new String(output.toByteArray(), StandardCharsets.UTF_8));
+            }
+        }
+    }
+
+    @Override
+    protected void doStop() throws Exception {
+        super.doStop();
+        httpClient.close();
```

Comment:
Isn't better to have `httpClient.close` before `super.doStop()`?

Reply:
I'm seeing file and seda behave differently there. I think `super.doStop()` should be first so it logs the stop operation. If the client throws an exception, at least you get the log before.